### ISSUE TITLE
Adicionar Input com autosuggestion

### ISSUE THREAD

**Issue by uselessdev** [pagarme/pilot#943]
### Contexto
Atualmente não temos nenhum `input` que possibilita uma busca um filtragem baseado em uma lista de dados, como isso já foi discutido abri essa issue pra podermos discutir onde seria melhor criar esse componente e qual seria a melhor forma.

Na minha opinião tanto faz adicionar no **FormerKit** já que o componente não precisa ficar acoplado a nenhuma regra de negócio e adicionando no **FormerKit**, ou na **Pilot** visto que nosso componente de **Input** é genérico pra podermos fazer isso como fizemos com o `CurrencyInput`.

### Exemplo
A minha ideia é que o componente seja baseado [nessa lib](http://react-autosuggest.js.org/) evitando um retrabalho visto também que ela parece ser bem customízavel e também parece ser bastante acessível. 

### Descrição técnica
Possível interface pública:

```jsx
<AutoSuggestionInput
  data={[
    { name: '', value: '' },
    { name: '', value: '' },
  ]}
  name="name"
  placeholder="placeholder"
  onChange={() => null}
  {...props} # outras props do componente.
/>
```

### Requisitos e Critérios de Aceite
- [ ] Ao digitar deve sugerir valores que estejam na lista `data`

**Comment by Deivis** [comment]
@uselessdev  mudei o nome no exemplo para `SuggestionInput` por que a parte `auto` quem vai fazer é o pai dele, o qual buscará os valores e dará as sugestões. Também adicionei um evento para quando um item for selecionado, só para diferenciar do `onChange` isso evita que o calback do pai que busca dados possa vir a ser chamado sem querer.

**Comment by uselessdev** [comment]
@Deivis a ideia é que o componente em si não fique responsável por buscar dados, mas receba por props a lista que for `searchable`

**Comment by Deivis** [comment]
Adicionei mais os seguintes campos:

 - `listSize`: que recebe um dos valores predefinidos(`small`, `medium` e `large`) e diz qual será a altura da lista, esta altura virá do css, é uma interfaca semelhante a do tamanho do `input`.

 - `itemsRenderer`: esta função serve pra renderizar os itens da lista de forma customizada, caso seja necessário dar destaque a algum item, ou simplesmente modificar o estilo utilizado, ela deve retornar um `element`.

**Comment by uselessdev** [comment]
@Deivis acho que `listSize` um pouco confuso pra definir tamanho da lista pode gerar confusão e acabar cortando items da lista, não seria melhor limitarmos por número de items exibidos :thinking: ?

**Comment by Deivis** [comment]
Não, quem limita o número de itens é o pai, e podemos cirar um scroll se necessário, isso é só para evitar listas gigantes que quebrem a tela.

**Comment by uselessdev** [comment]
Mas `listSize` ainda fica confuso, eu não entendi qual o propósito dessa `prop` ou como ela deveria funcionar

**Comment by Deivis** [comment]
Ela só adiciona um estilo X ou Y na lista que determina o height, o nome não tá bão  isso pode ser melhorado, talvez `listHeight`.

**Comment by Deivis** [comment]
Expliquei mau mesmo, é uma altura máxima, predefinida por nosso estilo, e caso tenhamos muitos itens colocamos um scroll ou algo mais bonitinho. E a janelinha se adapta a lista caso ela seja menor.

**Comment by uselessdev** [comment]
Agora eu entendi :joy:

**Comment by Deivis** [comment]
Troquei o nome para `listMaxHeight`, mas ainda parece meio ruim ainda :grimacing:

**Comment by uselessdev** [comment]
Realmente ficou um pouco 'estranho' ainda o se for algo como `maxItens`?

**Comment by tisoap** [comment]
Tenho uma sugestão: o que acham de fazer a interface deste componente similar ao do [`<Dropdown />`](https://pagarme.github.io/former-kit/#dropdown)? Assim poderíamos tratar o `<SuggestionInput />` como um drop down pesquisável.

**Comment by tisoap** [comment]
Tenho uma sugestão: o que acham de fazer a interface deste componente similar ao do [`<Dropdown />`](https://pagarme.github.io/former-kit/#dropdown)? Assim poderíamos tratar o `<SuggestionInput />` como um drop down pesquisável.

@uselessdev @Deivis @rsmelo

**Comment by uselessdev** [comment]
Acho estranho, visto que ele praticamente seria um custom `select`

**Comment by uselessdev** [comment]
Atualmente a interface do componente só permite selecionar 1 opção da lista de sugestão ou digitar qualquer coisa.

O maior problema que vejo aqui é que vocês tem opções limitadas o que não é garantida por este componente, se a idea for ter um select que seja pesquisável acho que seria mais interessante utilizar outro componente de base pra fazer o wrap, talvez usar o [react-select](https://github.com/JedWatson/react-select).

**Comment by tisoap** [comment]
Mas ai não fica ruim adicionar mais uma dependência no projeto?

**Comment by uselessdev** [comment]
Sim fica, mas se for um requisito não há o que fazer

**Comment by tisoap** [comment]
Eu acho que dá para utilizar o `react-autosuggest` para ambos os casos, evitando adicionar mais uma dependência: Dois componentes, um de sugestão e outro de seleção, que são apenas abstrações diferentes da mesma lib. O que acha?

**Comment by tisoap** [comment]
Status update: tentei, mas não dá para usar o `react-autosuggest` para criar opções limitadas... :weary:

Vou testar uma implementação com o `react-select`. Obrigado por mencionar esta lib!

**Comment by uselessdev** [comment]
eu te entendo @tisoap :joy:
